### PR TITLE
Refactored stateful storage pallet unit tests to use new layout.

### LINKED ISSUE #1238: Unit Test Reorganization:  Update test layout for the stateful storage pallet

### PR DESCRIPTION
**Pull request by rlaferla**
# Goal
The goal of this PR is refactor the unit tests to use the new unit test layout.

Closes #1238

### CODE REVIEW

File: pallets/stateful-storage/src/runtime-api/test_constants.rs
Comment (aramikm) [inline pallets/stateful-storage/src/runtime-api/test_constants.rs:1]: why an empty file?

File: pallets/stateful-storage/src/test_common.rs [hunk 1]
```diff
@@ -28,3 +28,61 @@ pub mod constants {
 	pub const PAGINATED_APPEND_ONLY_SCHEMA: SchemaId = 107;
 	pub const UNDELEGATED_PAGINATED_SCHEMA: SchemaId = 108;
 }
+
+#[cfg(test)]
+pub mod test_constants {
```
Comment (aramikm) [inline pallets/stateful-storage/src/test_common.rs:33]: this is not a constant module more like a test utility module

File: pallets/stateful-storage/src/tests/child_tree_tests.rs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+			<StatefulChildTree>::try_read(&msa_id, pallet_name, storage_name_1, keys).unwrap();
+		assert_eq!(Some(val), read);
+	});
+}
+
+type TestKeyString = BoundedVec<u8, ConstU32<16>>;
+type TestKey = (TestKeyString, TestKeyString, u64);
+
+#[test]
+fn child_tree_iterator() {
```
Comment (aramikm) [inline pallets/stateful-storage/src/tests/child_tree_tests.rs:41]: this should be in child tree file

File: pallets/stateful-storage/src/tests/itemized_operations_tests.rs [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+
+	// act
+	let result = ItemizedOperations::<Test>::apply_item_actions(&page, &actions[..]);
+
+	// assert
+	assert_eq!(result, Err(PageError::PageSizeOverflow));
+}
+
+#[test]
+fn apply_delete_item_on_append_only_fails() {
```
Comment (aramikm) [inline pallets/stateful-storage/src/tests/itemized_operations_tests.rs:143]: I think this should be in apply_item_actions file and the test name might also need to change to use the convention

File: pallets/stateful-storage/src/tests/other_tests.rs [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+#[test]
+fn is_empty_true_for_empty_page() {
+	let page: ItemizedPage<Test> = create_itemized_page_from::<Test>(None, &[]);
+
+	assert_eq!(page.is_empty(), true);
+}
+
+#[test]
+fn apply_actions_on_signature_schema_fails() {
```
Comment (aramikm) [inline pallets/stateful-storage/src/tests/other_tests.rs:32]: This should be in apply_item_actions file and test name might need to change to be compatible with the convention

File: pallets/stateful-storage/src/tests/other_tests.rs [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+				NONEXISTENT_PAGE_HASH,
+				BoundedVec::try_from(actions1).unwrap(),
+			),
+			Error::<Test>::UnsupportedOperationForSchema
+		);
+	});
+}
+
+#[test]
+fn insert_page_fails_for_signature_schema() {
```
Comment (aramikm) [inline pallets/stateful-storage/src/tests/other_tests.rs:54]: need to be in upsert_page file

File: pallets/stateful-storage/src/tests/upsert_page_tests.rs [hunk 6]
```diff
@@ -0,0 +1,660 @@
...
+};
+use frame_support::{assert_err, assert_ok};
+#[allow(unused_imports)]
+use pretty_assertions::{assert_eq, assert_ne, assert_str_eq};
+use sp_core::{sr25519, Pair};
+use sp_runtime::MultiSignature;
+use sp_std::hash::Hasher;
+use twox_hash::XxHash64;
+
+pub fn hash_payload(data: &[u8]) -> PageHash {
```
Comment (aramikm) [inline pallets/stateful-storage/src/tests/upsert_page_tests.rs:22]: should be in test_common utility file

**Review by aramikm — COMMENTED** [review]
Added some comments and suggestions

File: pallets/stateful-storage/src/tests/other_tests.rs (hunk 5 above)
Comment (rlaferla) [reply pallets/stateful-storage/src/tests/other_tests.rs:54]: fixed

File: pallets/stateful-storage/src/tests/other_tests.rs (hunk 4 above)
Comment (rlaferla) [reply pallets/stateful-storage/src/tests/other_tests.rs:32]: I will change it to `apply_item_actions_on_signature_schema_fails()` and move it.

File: pallets/stateful-storage/src/tests/child_tree_tests.rs (hunk 2 above)
Comment (rlaferla) [reply pallets/stateful-storage/src/tests/child_tree_tests.rs:41]: hmm.  Isn't it in `child_tree_tests.rs` already?

File: pallets/stateful-storage/src/test_common.rs (hunk 1 above)
Comment (rlaferla) [reply pallets/stateful-storage/src/test_common.rs:33]: Changed to `test_utility`

File: pallets/stateful-storage/src/tests/itemized_operations_tests.rs (hunk 3 above)
Comment (rlaferla) [reply pallets/stateful-storage/src/tests/itemized_operations_tests.rs:143]: There are several.  You previously wrote: "the ones starting with parsing_item and applying_delete or applying_add are ItemizedOperations". Let me know if you want to keep them in this file or change to the apply_items_actions file.

File: pallets/stateful-storage/src/tests/child_tree_tests.rs (hunk 2 above)
Comment (aramikm) [reply pallets/stateful-storage/src/tests/child_tree_tests.rs:39]: oops for some reason I thought it was in apply_item_action. my bad.

File: pallets/stateful-storage/src/tests/itemized_operations_tests.rs (hunk 3 above)
Comment (aramikm) [reply pallets/stateful-storage/src/tests/itemized_operations_tests.rs:143]: If it was testing ` ItemizedOperations::<Test>::apply_item_actions` I would have kept it here

Comment (aramikm) [reply pallets/stateful-storage/src/tests/itemized_operations_tests.rs:143]: The distinction is what is triggered in the` // act` section of the test

Comment (rlaferla) [reply pallets/stateful-storage/src/tests/itemized_operations_tests.rs:143]: Therefore I will move `apply_item_actions_existing_page_with_stale_hash_should_fail()` and `apply_item_actions_with_signature_having_page_with_stale_hash_should_fail()` to `itemized_operations_tests.rs`.

**Review by aramikm — APPROVED** [review]
Looks great!

**Pull request merged**